static/js/src/advantage/subscribe/react/PurchaseModal.jsx

```diff
@@ -0,0 +1,319 @@
+
+  const {
+    data: pendingPurchase,
+    setPendingPurchaseID,
+    error: purchaseError,
+    isLoading: isPendingPurchaseLoading,
+  } = usePendingPurchase();
+
+  const paymentMethodMutation = registerPaymentMethod();
+  const purchaseMutation = makePurchase();
+
+  const initialValues = {
+    email: userInfo?.customerInfo?.email ?? "",
+    name: userInfo?.customerInfo?.name ?? "",
+    buyingFor: "organisation",
+    organisationName: userInfo?.accountInfo?.name ?? "",
+    address: userInfo?.customerInfo?.address?.line1 ?? "",
+    postalCode: userInfo?.customerInfo?.address?.postal_code ?? "",
+    country: userInfo?.customerInfo?.address?.country ?? "",
+    city: userInfo?.customerInfo?.address?.city ?? "",
+    usState: userInfo?.customerInfo?.address?.state ?? "",
+    CAProvince: userInfo?.customerInfo?.address?.state ?? "",
+    VATNumber: userInfo?.customerInfo?.taxID?.value ?? "",
+  };
+
+  useEffect(() => {
+    if (userInfo?.customerInfo?.defaultPaymentMethod) {
+      setStep(2);
+    }
+  }, [userInfo]);
+
+  useEffect(() => {
+    // the initial call was successful but it returned an error while polling the purchase status
+    if (purchaseError) {
+      if (
+        purchaseError.message.includes(
+          "We are unable to authenticate your payment method"
+        )
+      ) {
+        setError(
+          <>
+            We were unable to verify your credit card. Check the details and try
+            again. Contact{" "}
+            <a href="https://ubuntu.com/contact-us">Canonical sales</a> if the
+            problem persists.
+          </>
+        );
+      } else {
+        setError(
+          <>
+            We were unable to process the payment. Check the details and try
+            again. Contact{" "}
+            <a href="https://ubuntu.com/contact-us">Canonical sales</a> if the
+            problem persists.
+          </>
+        );
+      }
+      setStep(1);
+    }
+  }, [purchaseError]);
+
+  useEffect(() => {
+    if (pendingPurchase?.status === "done") {
+      // Remove the product selection from the local storage
+      localStorage.removeItem("ua-subscribe-state");
+
+      //redirect
+      if (window.isGuest) {
+        location.href = `/advantage/subscribe/thank-you?email=${encodeURIComponent(
+          pendingPurchase?.invoice?.customerEmail
+        )}`;
+      } else {
+        location.pathname = "/advantage";
+      }
+    }
+  }, [pendingPurchase]);
+
+  const onSubmit = (values, actions) => {
+    setError(null);
+    paymentMethodMutation.mutate(values, {
+      onSuccess: (data, variables) => {
+        window.accountId = data.accountId;
+        setStep(2);
+        queryClient.setQueryData(
+          "userInfo",
+          getUserInfoFromVariables(data, variables)
+        );
+
+        actions.setSubmitting(false);
+      },
+      onError: (error) => {
+        if (error.message === "email_already_exists") {
+          //Email already exists
```

Comment:
I think this is clear from the condition above

Reply:
:see_no_evil: Whoops that was a leftover from when the error message was cryptic!